### PR TITLE
Various tutorial/example improvements

### LINKED ISSUE #15419: Use always preset DirichletBC in tutorials and examples
## Reason
We should use the new preferred method of presetting all DirichletBCs for all modules/tutorials/examples. This has to be set manually until we change the default.

## Design
Set `use_legacy_dirichlet_bc = false` in the tutorials and examples

## Impact
Minimal

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #15419

We should get this in before the workshop so that we get rid of this:
```
LEGACY MODES ENABLED:
 Default for parameter preset = false for all DirichletBC and ADDirichletBC objects.
 Set use_legacy_dirichlet_bc = false in the application to change to the preferred behavior.
```

### CODE REVIEW

**Review by lindsayad — DISMISSED** [review]

**Comment by loganharbour** [comment]
I approve of the changes here that aren't my own

**Review by cticenhour — DISMISSED** [review]

File: tutorials/darcy_thermo_mech/doc/content/systems/kernels.md [hunk 1]
```diff
@@ -36,39 +36,10 @@ Current quadrature point index
...
 !---
 
-## Kernel Base Classes
+## Kernel Base Class
 
 | Base | Override | Use |
 | :- | :- | :- |
 | Kernel\\ +ADKernel+ | computeQpResidual | Use when the term in the [!ac](PDE) is multiplied by both the test function and the gradient of the test function (`_test` and `_grad_test` must be applied) |
-| KernelValue\\ +ADKernelValue+ | precomputeQpResidual | Use when the term computed in the [!ac](PDE) is only multiplied by the test function (do not use `_test` in the override, it is applied automatically) |
-| KernelGrad\\ +ADKernelGrad+ | precomputeQpResidual | Use when the term computed in the [!ac](PDE) is only multiplied by the gradient of the test function (do not use `_grad_test` in the override, it is applied automatically) |
```
Comment (aeslaughter) [inline tutorials/darcy_thermo_mech/doc/content/systems/kernels.md:45]: I don't want to get rid of these. This is good information to present so when they run across them they understand what they are. I also will use them for the LaplaceYoug.

**Review by aeslaughter — CHANGES_REQUESTED** [review]

File: tutorials/darcy_thermo_mech/doc/content/systems/kernels.md (hunk 1 above)
Comment (cticenhour) [reply tutorials/darcy_thermo_mech/doc/content/systems/kernels.md]: I reverted this change. As we discussed in the video call, I'll speak about `ADDiffusion` as a standalone example of usage and brief intro to those specialized classes. Then we can move into `DarcyPressure` as an example of standard `ADKernel`

**Review by aeslaughter — DISMISSED** [review]

**Review by aeslaughter — DISMISSED** [review]

**Comment by loganharbour** [comment]
Only failure on last job (https://civet.inl.gov/job/542832/) now passing locally:

```
﻿(moose) [harbour][~/projects/moose/examples]> make -j 12 test                                                                            
ex01 ... OK
ex02 ... OK
ex03 ... OK
ex04 ... OK
ex04 ... OK
ex04 ... OK
ex04 ... OK
ex05 ... OK
Checking if header needs updating: /Users/harblh/projects/moose/framework/include/base/MooseRevision.h...
ex06 ... OK
Checking if header needs updating: /Users/harblh/projects/moose/framework/include/base/MooseRevision.h...
ex07 ... OK
ex07 ... OK
ex08 ... OK
Checking if header needs updating: /Users/harblh/projects/moose/framework/include/base/MooseRevision.h...
ex09 ... OK
ex10 ... OK
ex11 ... OK
ex12 ... OK
ex13 ... OK
ex14 ... OK
ex14 ... OK
ex14 ... OK
Rebuilding symlinks in /Users/harblh/projects/moose/examples/ex15_actions/build/header_symlinks
Creating Unity Directory /Users/harblh/projects/moose/examples/ex15_actions/build/unity_src
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex15_actions/src/base/ExampleApp.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex15_actions/src/main.C...
Linking Library /Users/harblh/projects/moose/examples/ex15_actions/test/lib/libex15_test-opt.la...
Creating Unity /Users/harblh/projects/moose/examples/ex15_actions/build/unity_src/kernels_Unity.C
Creating Unity /Users/harblh/projects/moose/examples/ex15_actions/build/unity_src/actions_Unity.C
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex15_actions/build/unity_src/kernels_Unity.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex15_actions/build/unity_src/actions_Unity.C...
Linking Library /Users/harblh/projects/moose/examples/ex15_actions/lib/libex15-opt.la...
Linking Executable /Users/harblh/projects/moose/examples/ex15_actions/ex15-opt...
ex15 ... OK
Rebuilding symlinks in /Users/harblh/projects/moose/examples/ex16_timestepper/build/header_symlinks
Creating Unity Directory /Users/harblh/projects/moose/examples/ex16_timestepper/build/unity_src
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex16_timestepper/src/base/ExampleApp.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex16_timestepper/src/main.C...
Linking Library /Users/harblh/projects/moose/examples/ex16_timestepper/test/lib/libex16_test-opt.la...
Creating Unity /Users/harblh/projects/moose/examples/ex16_timestepper/build/unity_src/kernels_Unity.C
Creating Unity /Users/harblh/projects/moose/examples/ex16_timestepper/build/unity_src/materials_Unity.C
Creating Unity /Users/harblh/projects/moose/examples/ex16_timestepper/build/unity_src/timesteppers_Unity.C
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex16_timestepper/build/unity_src/kernels_Unity.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex16_timestepper/build/unity_src/materials_Unity.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex16_timestepper/build/unity_src/timesteppers_Unity.C...
Linking Library /Users/harblh/projects/moose/examples/ex16_timestepper/lib/libex16-opt.la...
Linking Executable /Users/harblh/projects/moose/examples/ex16_timestepper/ex16-opt...
ex16 ... OK
Rebuilding symlinks in /Users/harblh/projects/moose/examples/ex17_dirac/build/header_symlinks
Creating Unity Directory /Users/harblh/projects/moose/examples/ex17_dirac/build/unity_src
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex17_dirac/src/base/ExampleApp.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex17_dirac/src/main.C...
Linking Library /Users/harblh/projects/moose/examples/ex17_dirac/test/lib/libex17_test-opt.la...
Creating Unity /Users/harblh/projects/moose/examples/ex17_dirac/build/unity_src/kernels_Unity.C
Creating Unity /Users/harblh/projects/moose/examples/ex17_dirac/build/unity_src/dirackernels_Unity.C
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex17_dirac/build/unity_src/kernels_Unity.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex17_dirac/build/unity_src/dirackernels_Unity.C...
Linking Library /Users/harblh/projects/moose/examples/ex17_dirac/lib/libex17-opt.la...
Linking Executable /Users/harblh/projects/moose/examples/ex17_dirac/ex17-opt...
ex17 ... OK
Rebuilding symlinks in /Users/harblh/projects/moose/examples/ex18_scalar_kernel/build/header_symlinks
Creating Unity Directory /Users/harblh/projects/moose/examples/ex18_scalar_kernel/build/unity_src
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex18_scalar_kernel/src/base/ExampleApp.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex18_scalar_kernel/src/main.C...
Linking Library /Users/harblh/projects/moose/examples/ex18_scalar_kernel/test/lib/libex18_test-opt.la...
Creating Unity /Users/harblh/projects/moose/examples/ex18_scalar_kernel/build/unity_src/scalarkernels_Unity.C
Creating Unity /Users/harblh/projects/moose/examples/ex18_scalar_kernel/build/unity_src/bcs_Unity.C
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex18_scalar_kernel/build/unity_src/scalarkernels_Unity.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex18_scalar_kernel/build/unity_src/bcs_Unity.C...
Linking Library /Users/harblh/projects/moose/examples/ex18_scalar_kernel/lib/libex18-opt.la...
Linking Executable /Users/harblh/projects/moose/examples/ex18_scalar_kernel/ex18-opt...
ex18 ... OK
ex18 ... OK
Rebuilding symlinks in /Users/harblh/projects/moose/examples/ex19_dampers/build/header_symlinks
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex19_dampers/src/base/ExampleApp.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex19_dampers/src/main.C...
Linking Library /Users/harblh/projects/moose/examples/ex19_dampers/test/lib/libex19_test-opt.la...
Linking Library /Users/harblh/projects/moose/examples/ex19_dampers/lib/libex19-opt.la...
Linking Executable /Users/harblh/projects/moose/examples/ex19_dampers/ex19-opt...
ex19 ... OK
Rebuilding symlinks in /Users/harblh/projects/moose/examples/ex20_user_objects/build/header_symlinks
Creating Unity Directory /Users/harblh/projects/moose/examples/ex20_user_objects/build/unity_src
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex20_user_objects/src/base/ExampleApp.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex20_user_objects/src/main.C...
Linking Library /Users/harblh/projects/moose/examples/ex20_user_objects/test/lib/libex20_test-opt.la...
Creating Unity /Users/harblh/projects/moose/examples/ex20_user_objects/build/unity_src/kernels_Unity.C
Creating Unity /Users/harblh/projects/moose/examples/ex20_user_objects/build/unity_src/materials_Unity.C
Creating Unity /Users/harblh/projects/moose/examples/ex20_user_objects/build/unity_src/userobjects_Unity.C
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex20_user_objects/build/unity_src/kernels_Unity.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex20_user_objects/build/unity_src/materials_Unity.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex20_user_objects/build/unity_src/userobjects_Unity.C...
Linking Library /Users/harblh/projects/moose/examples/ex20_user_objects/lib/libex20-opt.la...
Linking Executable /Users/harblh/projects/moose/examples/ex20_user_objects/ex20-opt...
ex20 ... OK
Rebuilding symlinks in /Users/harblh/projects/moose/examples/ex21_debugging/build/header_symlinks
Creating Unity Directory /Users/harblh/projects/moose/examples/ex21_debugging/build/unity_src
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex21_debugging/src/base/ExampleApp.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex21_debugging/src/main.C...
Linking Library /Users/harblh/projects/moose/examples/ex21_debugging/test/lib/libex21_test-opt.la...
Creating Unity /Users/harblh/projects/moose/examples/ex21_debugging/build/unity_src/kernels_Unity.C
Creating Unity /Users/harblh/projects/moose/examples/ex21_debugging/build/unity_src/materials_Unity.C
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex21_debugging/build/unity_src/kernels_Unity.C...
Compiling C++ (in opt mode) /Users/harblh/projects/moose/examples/ex21_debugging/build/unity_src/materials_Unity.C...
Linking Library /Users/harblh/projects/moose/examples/ex21_debugging/lib/libex21-opt.la...
Linking Executable /Users/harblh/projects/moose/examples/ex21_debugging/ex21-opt...
ex21 .. N/A
```

**Review by aeslaughter — APPROVED** [review]

**Comment by aeslaughter** [comment]
Merging to get website update for workshop, see the comment above: https://github.com/idaholab/moose/pull/15420#issuecomment-641020984.

**Pull request merged**